### PR TITLE
Error Handling Server

### PR DESCRIPTION
**Pull request by jstavh**
Printing the error for networking to GitHub failing. If this code fails to reach the URL "https://raw.githubusercontent.com/Granola-Team/mina-ledger/main/{network}/{hash}.json" due to a networking issue, it will print an error message that includes the details of the error. This is because of the following lines of added code:
```
                if let Err(error) = ledger_response {
                    // Print the error message to the console
                    eprintln!("Failed to fetch ledger from URL: {error:?}");

                    // Return an error
                    return Err(error.into());
                }

### CODE REVIEW

**Comment by robinbb** [comment]
@jstavh What remains on this PR to get it out of draft status?

**Comment by jstavh** [comment]
If this code fails to reach the URL "https://raw.githubusercontent.com/Granola-Team/mina-ledger/main/{network}/{hash}.json" due to a networking issue, it will print an error message that includes the details of the error. This is because of the following lines of added code:
```
                if let Err(error) = ledger_response {
                    // Print the error message to the console
                    eprintln!("Failed to fetch ledger from URL: {error:?}");

                    // Return an error
                    return Err(error.into());
                }

**Comment by jstavh** [comment]
> @jstavh What remains on this PR to get it out of draft status?

@robinbb I accidentally deleted the comment I wrote saying I'm working on it. Ready for review now though.

File: server/src/models/ledger.rs [hunk 1]
```diff
@@ -51,9 +51,16 @@ impl Ledger {
...
+                    .with_context(|| format!("failed to fetch ledger from URL: {ledger_url}"));
+
+                if let Err(error) = ledger_response {
+                    eprintln!("Failed to fetch ledger from URL: {error:?}");
+
+                    return Err(error.into());
+                }
 
                 let ledger_bytes = ledger_response
+                    .expect("Failed to fetch ledger response")
```
Comment (52) [inline server/src/models/ledger.rs:63]: `.expect` can panic, even though this case it won't, and is generally discouraged.

File: server/src/models/ledger.rs [hunk 2]
```diff
@@ -51,9 +51,16 @@ impl Ledger {
 
                 let ledger_response = reqwest::get(&ledger_url)
                     .await
-                    .with_context(|| format!("failed to fetch ledger from URL: {ledger_url}"))?;
+                    .with_context(|| format!("failed to fetch ledger from URL: {ledger_url}"));
+
+                if let Err(error) = ledger_response {
+                    eprintln!("Failed to fetch ledger from URL: {error:?}");
```
Comment (52) [inline server/src/models/ledger.rs:57]: Use `tracing::error!` instead.

**Review by 52 — CHANGES_REQUESTED** [review]
So this is a bit of a tough one, generally I'm opposed to localized handling of errors this way. Ideally, they should all propagate up & be handled in `error.rs`.

Also technically, this seems like we're duplicating things:
I haven't reproduced this, but the error on Line 54 in `ledger.rs` should already do what you want. 

Have you checked this @jstavh?

File: server/src/models/ledger.rs (hunk 1 above)
Comment (robinbb) [reply server/src/models/ledger.rs:63]: What's the alternative? unwrap()?

Comment (52) [reply server/src/models/ledger.rs]: @robinbb In this case yes, but the code has changed since, so I'm resolving this

**Comment by jstavh** [comment]
The error conversion between `error::Error` and `reqwest::Error` is currently missing the From trait. The previous code addition and approach is cleaner and more compatible for error handling.

**Review by robinbb — APPROVED** [review]

**Pull request merged**